marshmallow/schema.py

```diff
@@ -620,18 +620,15 @@ def _do_load(
         else:
             processed_data = data
         if not errors:
-            try:
-                result = unmarshal(
-                    processed_data,
-                    self.fields,
-                    many=many,
-                    partial=partial,
-                    unknown=unknown,
-                    dict_class=self.dict_class,
-                    index_errors=self.opts.index_errors,
-                )
-            except ValidationError as error:
-                result = error.data
+            result = unmarshal(
+                processed_data,
+                self.fields,
+                many=many,
+                partial=partial,
+                unknown=unknown,
+                dict_class=self.dict_class,
+                index_errors=self.opts.index_errors,
+            )
             self._invoke_field_validators(unmarshal, data=result, many=many)
             errors = unmarshal.errors
             # Run schema-level validation.
```

Comment:
`_invoke_schema_validators()` is calling `unmarshal.run_validator()` and expecting `ValidationError` even though `unmarshal.run_validator()` catches `ValidationError` and does not raise it. It looks like all the error collecting/raising in `_invoke_schema_validators()` is dead code. All those errors build up in the error store and get raised at the end of `_do_load()`.